### ISSUE TITLE
`indent-binary-ops`: false positive after chains on new line

### ISSUE THREAD

**Issue by Simbiat** [eslint-stylistic/eslint-stylistic#1209]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

1. Have code like this
```typescript
const anchor = document.getElementById('anchor');
if (
  (anchor && (anchor.getAttribute('href')
                    ?.startsWith('#tab_name_')) === false)
  && anchor.tabIndex === 0
) {
  console.log('test');
}
```
2. Have `@stylistic/indent-binary-ops` rule enabled

**Expected**:
No errors\warnings found (aside from console.log)

**Actual**:
Whitespace on line 5 is identified as violation of `@stylistic/indent-binary-ops`

I tried to prepare reproduction on Stackblitz, but it `pnpm run play:lint` fails with `Missing script: play:lint` and `pnpm run lint ./src/index.ts` fails with
```
❯ pnpm run lint ./src/index.ts

> eslint-stylistic-repro@ lint /home/projects/vbwijrzvgv.github
> eslint . "./src/index.ts"

TypeError: Cannot read properties of undefined (reading '0')
    at Module.enableCompileCache (node:internal/modules/helpers:177:2181)
    at Object.eval (/home/projects/vbwijrzvgv.github/node_modules/.pnpm/eslint@9.39.4/node_modules/eslint/bin/eslint.js:13:25)
    at Module._compile (node:internal/modules/cjs/loader:160:18351)
    at Module._extensions..js (node:internal/modules/cjs/loader:160:19092)
    at Module.load (node:internal/modules/cjs/loader:160:16623)
    at Module._load (node:internal/modules/cjs/loader:160:14286)
    at TracingChannel.traceSync (node:diagnostics_channel:26:3678)
    at wrapModuleLoad (node:internal/modules/cjs/loader:160:3949)
    at Module.executeUserEntryPoint (node:internal/modules/run_main:179:2179)

Node.js v22.22.0
 ELIFECYCLE  Command failed with exit code 1.
```

Hope above reproduction steps are enough.

### Reproduction

In description

### System Info

```shell
System:
    OS: Windows 11 10.0.26200
    CPU: (16) x64 AMD Ryzen 7 9800X3D 8-Core Processor           
    Memory: 23.03 GB / 61.65 GB
  Binaries:
    Node: 24.15.0 - C:\Program Files\nodejs\node.EXE
    npm: 11.12.1 - C:\Users\simbi\Coding\simbiat.dev\node_modules\.bin\npm.EXE
    bun: 1.3.13 - C:\Users\simbi\.bun\bin\bun.EXE
  npmPackages:
    @stylistic/eslint-plugin: ^5.10.0 => 5.10.0 
    @typescript-eslint/parser: ^8.59.2 => 8.59.2 
    eslint: ^9.39.4 => 9.39.4
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
